src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -659,7 +660,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 {
                     countOfExplicitNames++;
 
-                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.UnderlyingMemberName(i))
+                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.TupleMemberName(i))
```

Comment:
The real reason for "reserving" the name is that it should not clash with what already exists per our own definition of the Well Known Members of this type.

The issue with "Rest" is that it would not clash in any position because of the forced nesting of item N8. 

Perhaps we should actually allow it at any position since there is no possible clash?